src/Mod/Sketcher/Gui/CommandCreateGeo.cpp

```diff
@@ -267,24 +301,57 @@ class DrawSketchHandlerLine: public DrawSketchHandler
     {
         if (Mode==STATUS_SEEK_First) {
             setPositionText(onSketchPos);
+
             if (seekAutoConstraint(sugConstr1, onSketchPos, Base::Vector2d(0.f,0.f))) {
                 renderSuggestConstraintsCursor(sugConstr1);
                 return;
             }
+            if (toolWidgetManager.isParameterSet(0) + toolWidgetManager.isParameterSet(1) == 2) {
+                pressButton(onSketchPos);
+                releaseButton(onSketchPos);
+                mouseMove(onSketchPos);

```

Comment:
recursive calls to a function are often not a good idea. It is not intrinsically wrong. There is generally the risk of infinite recursion, also when maintaining the code. I will have a look to see if we can avoid this.